js/scripts-2.js

```diff
@@ -0,0 +1,230 @@
+    }
+    map.fitBounds(bounds);
+}
+
+/*Creating Markers to display them on the Google map */
+function createMarker(latlng, nameSupermarket, addressSupermarket, phoneSupermarket, position) {
+    let html = `
+        <div class="store-info-window">
+            <div class="store-info-name">
+                ${nameSupermarket}
+            </div>
+            <div class="store-info-status">
+            Open until 8:00 PM
+            </div>
+            <div class="store-info-address">
+                <div class="circle">
+                    <i class="fa fa-location-arrow"></i>
+                </div>
+                ${addressSupermarket}
+            </div>
+            <div class="store-info-phone">
+                <div class="circle">
+                    <i class="fa fa-phone"></i>
+                </div>
+                ${phoneSupermarket}
+            </div>
+        </div>
+        `;
+    let marker = new google.maps.Marker({
+        map: map,
+        position: latlng,
+        label: `${position}`
+    });
+
+    google.maps.event.addListener(marker, 'click', function() {
+        infoWindow.setContent(html);
+        infoWindow.open(map, marker);
+    } )
+
+    markers.push(marker);
+
+};
+
+/*Triggering Info Window on the google map */
+function setOnClickListener() {
+    const storeElements = document.querySelectorAll('.store-container');
+    storeElements.forEach(function(elem, index){
+        elem.addEventListener('click', function(){
+            google.maps.event.trigger(markers[index], 'click');
+        })
+    });
+}
+
+// function searchstores(data) {
+//     // let foundStores = [];
+//     let inputValue = document.getElementById('inputValue').value;
+//     // console.log(inputValue);
+//     for (let i = 0; i < data.markets.length; i++) {
+//     }
+// };
+
+    function searchByPlace()  {
+        //var foundStores = [];
+        //Creating the searchbox
+        let input = document.getElementById('inputValue');
+        let autocomplete = new google.maps.places.Autocomplete(input);
+        autocomplete.bindTo('bounds', map)
+        // Listen for the event fired when the user selects a prediction and retrieve
+        // more details for that place.
+          autocomplete.addListener("places_changed", function () {
+            //infoWindow.close();
+            processButtonSearch();
+            marker.setVisible(false);
+            let place = autocomplete.getPlaces();
+            if (!place.geometry) {
+                window.alert("Autocomplete's rturned place contains no geomtery");
+                return;
+            }
+            //if the place has a geometry,then present it on a map
+            if(place.geometry.viewport) {
+                map.fitBounds(place.geometry.viewport);
+            } else {
+                map.setCenter(place.geometry.location);
+                map.setZoom(17);
+            }
+          });
+
+          showStoresMarker(data);
```

Comment:
where the `data` is coming from here? I think this will throw an error in the console.